### DISCUSSION TITLE
In VSCODE extension webview,  websocket client could not establish connection with remote websocket server

### DISCUSSION THREAD

**Discussion by jcyrss** [microsoft/vscode-discussions#693] [Extension Development QnA] [answered]
I found the websocket in the webview of vscode extension could not establish connection with a `remote` websocket server. 

The url I used is like `ws://192.168.5.33:8081`, Developer tools console shows info like 

```py
WebSocket connection to 'ws://192.168.5.33:8081/' failed: 
```

It does not look like it violates any Content Security Policy, for otherwise, that will be displayed.

Developer tools Network tab shows like this, 

![image](https://github.com/microsoft/vscode-discussions/assets/10496014/963cd2da-26c6-42c5-a113-361b6980f438)

![image](https://github.com/microsoft/vscode-discussions/assets/10496014/65b97d80-8fa5-40f2-ae67-425f54256547)

![image](https://github.com/microsoft/vscode-discussions/assets/10496014/a0128763-ae3e-4e1d-a215-4a59a6a90cdc)

Just a request, no response. I'm sure websocket server works well, because I run the same code `in browser` instead of vscode webview extension, to connect with that remote server, and found it works well.

Also that extension webview works well with a **local** websocket server, like `ws://127.0.0.1:8081`

I run tcpdump on server host, trying to connect to ws server in vscode extension webview did not get any captured package. While it did when I connected in the browser.

So it seems vscode does not allow my webview code to contact with remote websocket server.

Does vscode have any restriction on **extension webview websocket** to connect with A **REMOTE** server?

**Comment by r3m0t** [comment]
Open the Developer Tools from Extension Development Host, there will be information there e.g. if it's blocked by CSP.

**Comment by jcyrss** [comment]
> Open the Developer Tools from Extension Development Host, there will be information there e.g. if it's blocked by CSP.

I just updated the description to make it more clear. It seems not blocked by CSP.

**Reply by r3m0t** [reply]
Any idea why the port is different between 58888 and 8081?

Are there response headers shown below the request headers? 

If you press x on the request details panel is there more info?

**Reply by jcyrss** [reply]
sorry, they were 2 different tries with different remote websocket port. I mismatch the screenshots. But it failed in both cases in the same way.

And there was no any reponse shown below

Please see the screenshots in the updated description above.

**Comment by jcyrss** [comment, marked as answer]
It turns out due to the proxy in vscode set long time ago. By turning the proxy off, everything goes well both in webview and in extension native.
